tests/Festung/Vault/VaultHandlerSpec.hs

```diff
@@ -78,13 +78,37 @@ spec = do
                 Left  _            -> expectationFailure "is not CouldNotOpen"
                 Right _            -> expectationFailure "is not an error"
 
-    describe "parametrizedQuery" $
+    describe "parametrizedQuery" $ do
         it "Parametrizes the query" $ withHandler_ $ \ opener vault -> do
             let query' = query opener vault
                 parametrizedQuery' = parametrizedQuery opener vault
 
-            query' "CREATE TABLE foo(bar int)"
-            parametrizedQuery' "INSERT INTO foo(bar) VALUES (?), (?)" [P.IntValue 1, P.IntValue 2]
-            dat <- noError =<< parametrizedQuery' "SELECT bar FROM foo WHERE bar < ?" [P.IntValue 2]
-            P.headers dat `shouldBe` [("bar", Just "int")]
+            noError =<< query' "CREATE TABLE foo(bar integer)"
+            noError =<< parametrizedQuery' "INSERT INTO foo(bar) VALUES (?)"  (P.Params [P.IntValue 1])
+            dat <- noError =<< query' "SELECT bar FROM foo WHERE bar"
+            P.headers dat `shouldBe` [("bar", Just "integer")]
             P.rows    dat `shouldBe` [[P.IntValue 1]]
+        
+        it "Parametrizes the query with bulk params" $ withHandler_ $ \ opener vault -> do
+            let query' = query opener vault
+                parametrizedQuery' = parametrizedQuery opener vault
+
+            noError =<< query' "CREATE TABLE foo(bar integer)"
+            noError =<< parametrizedQuery' "INSERT INTO foo(bar) VALUES (?)"  
+                (P.BulkParams [[P.IntValue 1], [P.IntValue 2], [P.IntValue 3]])
+            dat <- noError =<< query' "SELECT bar FROM foo"
+            P.headers dat `shouldBe` [("bar", Just "integer")]
+            P.rows    dat `shouldBe` [[P.IntValue 1], [P.IntValue 2], [P.IntValue 3]]
+
+        it "Stops inserting if fails in the middle"  $ withHandler_ $ \ opener vault -> do
+            pendingWith "Find out why insert not fails with null or invalid types"
```

Comment:
SQLite uses an unconventional dynamic [type system](https://www.sqlite.org/datatype3.html), which sometimes is confusing when you're used to a more rigid type system. Consider this example
```
sqlite> CREATE TABLE foo (bar INT);
sqlite> INSERT INTO foo VALUES ("a");
sqlite> SELECT bar, typeof(bar) FROM foo;
bar         typeof(bar)
----------  -----------
a           text
```
The storage class that you used in the table declaration is only the recommended storage class, but you still can insert whatever values you want into that column